DPG/Tasks/AOTTrack/MonitorFilterBit.cxx

```diff
@@ -114,45 +114,44 @@ struct CheckFilterBit {
   }
   PROCESS_SWITCH(CheckFilterBit, processData, "process data", true);
 
-    template <typename T> int isFromStrangeDecay(const T& particlesMC,const typename T::iterator& particle){
+  template <typename T>
+  int isFromStrangeDecay(const T& particlesMC, const typename T::iterator& particle)
+  {
 
     std::vector<std::vector<int64_t>> arrayIds{};
     std::vector<int64_t> initVec{particle.globalIndex()};
     arrayIds.push_back(initVec);
-    
-    int stage=0;
-    int strangeness=0;
-    while(stage<4 && arrayIds[stage].size()>0 && strangeness==0){
+
+    int stage = 0;
+    int strangeness = 0;
+    while (stage < 4 && arrayIds[stage].size() > 0 && strangeness == 0) {
```

Comment:
I do not understand why you are adding the condition `stage < 4` here. I guess it is a custom value (i.e. you check if a particle comes from a strange decay up to 4 stages above). Also, I see `arrayIds[stage].size() > 0` dangerous, because it might happen that you never call `arrayIds.push_back(arrayIdsStage);` because e.g. you `break;` before, and you might have unpredictable behaviours for `arrayIds[stage]`.
What about writing `while (stage < arrayIds.size() && strangeness == 0)` ?

Let me know if I'm fully misunderstanding something.